In [1]:
# Filtering the 6k Indian food dataset 
# This code can be integrated into dev.ipynb later if needed
import pandas as pd

In [2]:
# Importing dataset into the workspace
globals()['IndianRecipes'] = pd.read_csv("IndianFoodDatasetCSV.csv")
# Displaying
IndianRecipes

Srno                                         RecipeName  \
0         1                               Masala Karela Recipe   
1         2  टमाटर पुलियोगरे रेसिपी - Spicy Tomato Rice (Re...   
2         3  Ragi Semiya Upma Recipe - Ragi Millet Vermicel...   
3         4  Gongura Chicken Curry Recipe - Andhra Style Go...   
4         5  आंध्रा स्टाइल आलम पचड़ी रेसिपी - Adrak Chutney ...   
...     ...                                                ...   
6866  14073  गोअन मशरुम जकुटी रेसिपी - Goan Mushroom Xacuti...   
6867  14107  शकरकंदी और मेथी का पराठा रेसिपी - Sweet Potato...   
6868  14165      Ullikadala Pulusu Recipe | Spring Onion Curry   
6869  14167  Kashmiri Style Kokur Yakhni Recipe-Chicken Coo...   
6870  14211              नवरंग दाल रेसिपी - Navrang Dal Recipe   

                                   TranslatedRecipeName  \
0                                  Masala Karela Recipe   
1                            Spicy Tomato Rice (Recipe)   
2     Ragi Semiya Upma Recipe - Ragi Millet Vermicel...   
3     Gongura Chicken Curry Recipe - Andhra Style Go...   
4     Andhra Style Alam Pachadi Recipe - Adrak Chutn...   
...                                                 ...   
6866                        Goan Mushroom Xacuti Recipe   
6867        Sweet Potato & Methi Stuffed Paratha Recipe   
6868      Ullikadala Pulusu Recipe | Spring Onion Curry   
6869  Kashmiri Style Kokur Yakhni Recipe-Chicken Coo...   
6870                                 Navrang Dal Recipe   

                                            Ingredients  \
0     6 Karela (Bitter Gourd/ Pavakkai) - deseeded,S...   
1     2-1/2 कप चावल - पका ले,3 टमाटर,3 छोटा चमच्च बी...   
2     1-1/2 cups Rice Vermicelli Noodles (Thin),1 On...   
3     500 grams Chicken,2 Onion - chopped,1 Tomato -...   
4     1 बड़ा चमच्च चना दाल,1 बड़ा चमच्च सफ़ेद उरद दाल,2...   
...                                                 ...   
6866  20 बटन मशरुम,2 प्याज - काट ले,1 टमाटर - बारीक ...   
6867  1 बड़ा चम्मच तेल,1 कप गेहूं का आटा,नमक - स्वाद ...   
6868  150 grams Spring Onion (Bulb & Greens) - chopp...   
6869  1 kg Chicken - medium pieces,1/2 cup Mustard o...   
6870  2 बड़े चम्मच हरी मूंग दाल,2 बड़े चम्मच सफ़ेद उरद ...   

                                  TranslatedIngredients  PrepTimeInMins  \
0     6 Karela (Bitter Gourd/ Pavakkai) - deseeded,S...              15   
1     2-1 / 2 cups rice - cooked, 3 tomatoes, 3 teas...               5   
2     1-1/2 cups Rice Vermicelli Noodles (Thin),1 On...              20   
3     500 grams Chicken,2 Onion - chopped,1 Tomato -...              15   
4     1 tablespoon chana dal, 1 tablespoon white ura...              10   
...                                                 ...             ...   
6866  20 बटन मशरुम,2 प्याज - काट ले,1 टमाटर - बारीक ...              15   
6867  1 बड़ा चम्मच तेल,1 कप गेहूं का आटा,नमक - स्वाद ...              30   
6868  150 grams Spring Onion (Bulb & Greens) - chopp...               5   
6869  1 kg Chicken - medium pieces,1/2 cup Mustard o...              30   
6870  2 बड़े चम्मच हरी मूंग दाल,2 बड़े चम्मच सफ़ेद उरद ...              10   

      CookTimeInMins  TotalTimeInMins  Servings               Cuisine  \
0                 30               45         6                Indian   
1                 10               15         3  South Indian Recipes   
2                 30               50         4  South Indian Recipes   
3                 30               45         4                Andhra   
4                 20               30         4                Andhra   
...              ...              ...       ...                   ...   
6866              45               60         4          Goan Recipes   
6867              60               90         4  North Indian Recipes   
6868              10               15         2                Andhra   
6869              45               75         4              Kashmiri   
6870              30               40         4  North Indian Recipes   

                  

In [3]:
# Source: https://celiac.org/gluten-free-living/what-is-gluten/sources-of-gluten/
glutenContainingIngredients = ["wheat","wheatberries","durum","emmer","semolina","spelt","farina","farro","graham","Rye","Barley","Triticale","Malt","Brewer's Yeast"]
part2 = ["pasta", "raviolis", "dumplings", "couscous", "gnocchi", "noodles", "ramen", "udon", "soba", "chow mein", "egg noodles", "croissant", "pita", "naan", "bagel", "flatbread", "cornbread", "potato bread", "muffins", "donuts", "rolls", "pretzels", "goldfish", "graham crackers", "cake","cakes", "cookies", "pie crusts", "brownies", "corn flakes", "rice puffs", "pancakes", "waffles", "french toast", "crepes", "biscuit","biscuits", "panko breadcrumbs", "soy sauce", "flour tortillas", "beer"]
glutenContainingIngredients += part2

glutenContainingIngredients

['wheat',
 'wheatberries',
 'durum',
 'emmer',
 'semolina',
 'spelt',
 'farina',
 'farro',
 'graham',
 'Rye',
 'Barley',
 'Triticale',
 'Malt',
 "Brewer's Yeast",
 'pasta',
 'raviolis',
 'dumplings',
 'couscous',
 'gnocchi',
 'noodles',
 'ramen',
 'udon',
 'soba',
 'chow mein',
 'egg noodles',
 'croissant',
 'pita',
 'naan',
 'bagel',
 'flatbread',
 'cornbread',
 'potato bread',
 'muffins',
 'donuts',
 'rolls',
 'pretzels',
 'goldfish',
 'graham crackers',
 'cake',
 'cakes',
 'cookies',
 'pie crusts',
 'brownies',
 'corn flakes',
 'rice puffs',
 'pancakes',
 'waffles',
 'french toast',
 'crepes',
 'biscuit',
 'biscuits',
 'panko breadcrumbs',
 'soy sauce',
 'flour tortillas',
 'beer']

In [4]:
print(IndianRecipes.describe())

               Srno  PrepTimeInMins  CookTimeInMins  TotalTimeInMins  \
count   6871.000000     6871.000000     6871.000000      6871.000000   
mean    4902.979042       28.585213       30.832339        59.417552   
std     3479.531022       81.042007       34.019694        88.699940   
min        1.000000        0.000000        0.000000         0.000000   
25%     1955.500000       10.000000       20.000000        30.000000   
50%     4304.000000       15.000000       30.000000        40.000000   
75%     7363.000000       20.000000       35.000000        55.000000   
max    14211.000000     2880.000000      900.000000      2925.000000   

          Servings  
count  6871.000000  
mean      5.611410  
std      26.221807  
min       1.000000  
25%       4.000000  
50%       4.000000  
75%       4.000000  
max    1000.000000  


In [5]:
IndianRecipes.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 6871 entries, 0 to 6870
Data columns (total 15 columns):
 #   Column                  Non-Null Count  Dtype 
---  ------                  --------------  ----- 
 0   Srno                    6871 non-null   int64 
 1   RecipeName              6871 non-null   object
 2   TranslatedRecipeName    6871 non-null   object
 3   Ingredients             6865 non-null   object
 4   TranslatedIngredients   6865 non-null   object
 5   PrepTimeInMins          6871 non-null   int64 
 6   CookTimeInMins          6871 non-null   int64 
 7   TotalTimeInMins         6871 non-null   int64 
 8   Servings                6871 non-null   int64 
 9   Cuisine                 6871 non-null   object
 10  Course                  6871 non-null   object
 11  Diet                    6871 non-null   object
 12  Instructions            6871 non-null   object
 13  TranslatedInstructions  6871 non-null   object
 14  URL                     6871 non-null   object
dtypes: i

In [6]:
f=0
c=0
for x,y,i in zip(IndianRecipes["TranslatedIngredients"],IndianRecipes["Srno"],range(1,6872)):
    ingredientsList = str(x).split(",")
    #print(ingredientsList) # For testing purposes
    for ingredient in ingredientsList:
        for glutenContainingIngredient in glutenContainingIngredients:
            if glutenContainingIngredient in ingredient:
                print("CONTAINS GLUTEN!",i,y,glutenContainingIngredient,ingredient)
                # For testing purposes
                IndianRecipes.drop(i-1,inplace=True)
                f = 1
                c += 1
                break;
        if f==1:
            break;
    if f==1:
        f=0
        continue;


CONTAINS GLUTEN! 10 10 Rye  1 teaspoon Rye
CONTAINS GLUTEN! 16 16 Rye  as per taste Rye
CONTAINS GLUTEN! 28 31 pasta 1 tablespoon Roasted tomato pasta sauce
CONTAINS GLUTEN! 66 74 noodles 250 grams Egg noodles - boiled and tossed in a teaspoon oil
CONTAINS GLUTEN! 82 91 soy sauce 1 tablespoon Dark soy sauce
CONTAINS GLUTEN! 84 93 wheat  1 cup wheat flour
CONTAINS GLUTEN! 85 94 Rye  1-1 / 2 teaspoon Rye
CONTAINS GLUTEN! 86 95 noodles 200 grams rice noodles
CONTAINS GLUTEN! 129 139 biscuit 8 Oreo biscuits - broken into pieces
CONTAINS GLUTEN! 134 144 pasta 1 tablespoon Roasted tomato pasta sauce
CONTAINS GLUTEN! 159 171 pasta 1 tablespoon Roasted tomato pasta sauce - or store bought
CONTAINS GLUTEN! 236 251 wheat  1 tablespoon wheat flour
CONTAINS GLUTEN! 253 271 Barley 3/4 cup Barley flour
CONTAINS GLUTEN! 259 277 Barley 1 cup Barley (seeds) - soaked for 2-6 hours
CONTAINS GLUTEN! 271 289 soy sauce 2 teaspoon Light soy sauce
CONTAINS GLUTEN! 347 372 wheat  1 cup wheat flour
CONTAINS GLU

In [7]:
# re-running to check if all gluten containing items have been removed
for x,y,i in zip(IndianRecipes["TranslatedIngredients"],IndianRecipes["Srno"],range(1,6872)):
    ingredientsList = str(x).split(",")
    #print(ingredientsList) # For testing purposes
    for ingredient in ingredientsList:
        for glutenContainingIngredient in glutenContainingIngredients:
            if glutenContainingIngredient in ingredient:
                print("CONTAINS GLUTEN!",i,y,glutenContainingIngredient,ingredient)
                # For testing purposes
                #IndianRecipes.drop((i-1))

In [8]:
IndianRecipes.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 6558 entries, 0 to 6870
Data columns (total 15 columns):
 #   Column                  Non-Null Count  Dtype 
---  ------                  --------------  ----- 
 0   Srno                    6558 non-null   int64 
 1   RecipeName              6558 non-null   object
 2   TranslatedRecipeName    6558 non-null   object
 3   Ingredients             6552 non-null   object
 4   TranslatedIngredients   6552 non-null   object
 5   PrepTimeInMins          6558 non-null   int64 
 6   CookTimeInMins          6558 non-null   int64 
 7   TotalTimeInMins         6558 non-null   int64 
 8   Servings                6558 non-null   int64 
 9   Cuisine                 6558 non-null   object
 10  Course                  6558 non-null   object
 11  Diet                    6558 non-null   object
 12  Instructions            6558 non-null   object
 13  TranslatedInstructions  6558 non-null   object
 14  URL                     6558 non-null   object
dtypes: i

In [9]:
c

313

A total of 313 rows(recipes) have been removed. This can be verified by comparing the difference in number of rows from the original dataframe(6871) and filtered dataframe(6558) = 313.